domains/client/domain-operator/src/bundle_processor.rs

```diff
@@ -72,30 +93,154 @@ where
     }
 }
 
-struct ReceiptValidator<Client> {
+struct ReceiptValidator<Client, Block, CBlock, CClient, Backend, E, ParentChain, ParentChainBlock> {
+    domain_id: DomainId,
     client: Arc<Client>,
+    fraud_proof_generator: FraudProofGenerator<Block, CBlock, Client, CClient, Backend, E>,
+    parent_chain: ParentChain,
```

Comment:
If these fields are not used in this PR then we should not introduce them in this PR, if necessary a TODO comment should be enough.

Reply:
Make sense.